// @botonic/react/lib/webchat/components/attachment.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Attachment = void 0;

var _react = _interopRequireDefault(require("react"));

var _attachmentIcon = _interopRequireDefault(require("../../assets/attachment-icon.svg"));

var _constants = require("../../constants");

var _common = require("./common");

var Attachment = function Attachment(_ref) {
  var onChange = _ref.onChange,
      accept = _ref.accept;
  return /*#__PURE__*/_react["default"].createElement(_common.IconContainer, {
    role: _constants.ROLES.ATTACHMENT_ICON
  }, /*#__PURE__*/_react["default"].createElement("label", {
    htmlFor: "attachment",
    style: {
      marginTop: 4
    }
  }, /*#__PURE__*/_react["default"].createElement(_common.Icon, {
    src: _attachmentIcon["default"]
  })), /*#__PURE__*/_react["default"].createElement("input", {
    type: "file",
    name: "file",
    id: "attachment",
    style: {
      display: 'none'
    },
    onChange: onChange,
    accept: accept
  }));
};

exports.Attachment = Attachment;
