// @botonic/react/lib/webchat/components/persistent-menu.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.PersistentMenu = exports.OpenedPersistentMenu = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _menuButton = _interopRequireDefault(require("../../assets/menuButton.svg"));

var _button = require("../../components/button");

var _constants = require("../../constants");

var _contexts = require("../../contexts");

var _hooks = require("../hooks");

var _common = require("./common");

var _templateObject, _templateObject2;

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var ButtonsContainer = _styledComponents["default"].div(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  z-index: 2;\n  width: 100%;\n  bottom: 0;\n  text-align: center;\n"])));

var OpenedPersistentMenu = function OpenedPersistentMenu(_ref) {
  var onClick = _ref.onClick,
      options = _ref.options,
      borderRadius = _ref.borderRadius;

  var _useComponentVisible = (0, _hooks.useComponentVisible)(true, onClick),
      ref = _useComponentVisible.ref,
      isComponentVisible = _useComponentVisible.isComponentVisible;

  var _useContext = (0, _react.useContext)(_contexts.WebchatContext),
      getThemeProperty = _useContext.getThemeProperty;

  var CustomPersistentMenu = getThemeProperty(_constants.WEBCHAT.CUSTOM_PROPERTIES.customPersistentMenu, undefined);
  var closeLabel = 'Cancel';

  try {
    closeLabel = options.filter(function (opt) {
      return opt.closeLabel !== undefined;
    })[0].closeLabel;
  } catch (e) {}

  return /*#__PURE__*/_react["default"].createElement("div", {
    ref: ref,
    role: _constants.ROLES.PERSISTENT_MENU
  }, isComponentVisible && CustomPersistentMenu ? /*#__PURE__*/_react["default"].createElement(CustomPersistentMenu, {
    onClick: onClick,
    options: options
  }) : /*#__PURE__*/_react["default"].createElement(ButtonsContainer, null, Object.values(options).map(function (e, i) {
    return e.label && /*#__PURE__*/_react["default"].createElement(_button.Button, {
      onClick: onClick,
      url: e.url,
      target: e.target,
      webview: e.webview,
      payload: e.payload,
      key: i
    }, e.label);
  }), /*#__PURE__*/_react["default"].createElement(_button.Button, {
    onClick: onClick,
    bottomRadius: borderRadius
  }, closeLabel)));
};

exports.OpenedPersistentMenu = OpenedPersistentMenu;

var IconContainer = _styledComponents["default"].div(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  align-items: center;\n  cursor: pointer;\n  padding: 18px;\n"])));

var PersistentMenu = function PersistentMenu(props) {
  return /*#__PURE__*/_react["default"].createElement(IconContainer, {
    role: _constants.ROLES.PERSISTENT_MENU_ICON,
    onClick: props.onClick
  }, /*#__PURE__*/_react["default"].createElement(_common.Icon, {
    src: _menuButton["default"]
  }));
};

exports.PersistentMenu = PersistentMenu;
var _default = OpenedPersistentMenu;
exports["default"] = _default;
